src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -98,8 +135,20 @@ public GetSecretValueResponse handleRequest(GetSecretValueRequest request) {
 
         @Override
         public CompletableFuture<GetSecretValueResponse> handleRequestAsync(GetSecretValueRequest request) {
-            // TODO: Limit the no of simultaneous requests using a separate threadpool with configurable size
-            return CompletableFuture.supplyAsync(() -> handleRequest(request));
+            if (cloudCallThreadPool == null) {
+                // This should never happen.
+                cloudCallThreadPool = new ThreadPoolExecutor(1, 1, 60, TimeUnit.SECONDS,
```

Comment:
lets either refactor or make the numbers constants since we're creating this in multiple places